Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -343,15 +353,21 @@ void FullscreenManager::exitFullscreen()
             INFO_LOG(identifier, "task - Cancelling pending fullscreen request.");
             m_pendingFullscreenElement = nullptr;
             m_pendingExitFullscreen = false;
+            promise->resolve();
             return;
         }
 
+        if (m_pendingPromise)
+            m_pendingPromise->reject(Exception { TypeError });
+
         // Notify the chrome of the new full screen element.
-        if (mode == ExitMode::Resize)
+        if (mode == ExitMode::Resize) {
+            m_pendingPromise = WTFMove(promise);
             page->chrome().client().exitFullScreenForElement(m_fullscreenElement.get());
-        else {
+        } else {
             finishExitFullscreen(document(), ExitMode::NoResize);
 
+            m_pendingPromise = WTFMove(promise);
```

Comment:
Instead of duplicating the `m_pendingPromise = WTFMove(promise);` line, could we do it once before the `if (mode == ExitMode::Resize)` ?